File: app/src/main/java/alex/com/githubchecker/models/api/PullRequest.java
package alex.com.githubchecker.models.api;

/**
 * Created by Alex on 11/17/2017.
 */

public class PullRequest extends BaseModel {

    private String title;
    private Integer number;
    private Commit head;
    private String created_at;
    private String diff_url;


    //Accessors methods
    public String getTitle() {
        return title;
    }

    public String getUserLogin() {
        if (head != null) {
            return head.getUserLogin();
        }
        return null;
    }

    public Integer getNumber() {
        return number;
    }

    public String getCreatedAt() {
        return created_at;
    }

    public String getDiffUrl() {
        return diff_url;
    }
}

File: app/src/main/java/alex/com/githubchecker/models/api/PullRequest.kt
package alex.com.githubchecker.models.api

import com.google.gson.annotations.SerializedName

/**
 * Created by Alex on 11/17/2017.
 */

class PullRequest {
    var id: Int? = null
    var title: String? = null
    var number: Int? = null
    val head: Commit? = null
    var created_at: String? = null
    var diff_url: String? = null
}
